docs/05_Advanced_Features/00_Get_started_with_MPU6500.md

```diff
@@ -0,0 +1,335 @@
+}
+```
+
+Such functions simply calls extern functions, that will be defined and implemented in `MPU-driver-wrapper.cpp` file, which contain the relevant business logic.
+
+So the content of such file can be:
+
+```C++
+#include <esp_log.h>
+#include <esp_err.h>
+
+#include <MPU.hpp>
+#include <I2Cbus.hpp>
+#include <mpu/math.hpp>
+
+
+static const char *TAG = "MPU-driver-wrapper";
+
+
+
+extern "C" MPU_t *mpu_get_new_driver_object() {
+    MPU_t *sensor = NULL;
+    esp_err_t res = ESP_OK;
+    
+    i2c0.begin(GPIO_NUM_18, GPIO_NUM_21, 400000);
+
+    sensor   = new MPU_t();
+    sensor->setBus(i2c0);
+    sensor->setAddr(mpud::MPU_I2CADDRESS_AD0_HIGH);
+
+    if ((res = sensor->testConnection()) != ESP_OK) {
+        ESP_LOGE(TAG, "Failed to connect to the MPU, error=%#X (%s)", res, esp_err_to_name(res));
+        vTaskDelay(pdMS_TO_TICKS(100));
+    }
+
+    ESP_ERROR_CHECK(sensor->initialize());
+
+    return sensor;
+}
+
+
+
+
+extern "C" esp_err_t mpu_accelerometer_read(MPU_t *MPU_driver, float *x, float *y, float *z) {
+    mpud::raw_axes_t accelRaw;
+    mpud::float_axes_t accelG;
+    
+    // Reading
+    MPU_driver->acceleration(&accelRaw);
+    // Converting
+    accelG = mpud::accelGravity(accelRaw, mpud::ACCEL_FS_4G);
+
+    *x = accelG.x;
+    *y = accelG.y;
+    *z = accelG.z;
+
+    return ESP_OK;
+}
+
+
+
+
+extern "C" esp_err_t mpu_gyroscope_read(MPU_t *MPU_driver, float *x, float *y, float *z) {
+    mpud::raw_axes_t gyroRaw;
+    mpud::float_axes_t gyroDPS;
+    
+    // Reading
+    MPU_driver->rotation(&gyroRaw);
+    // Converting
+    gyroDPS = mpud::gyroDegPerSec(gyroRaw, mpud::GYRO_FS_500DPS);
+
+    *x = gyroDPS[0];
+    *y = gyroDPS[1];
+    *z = gyroDPS[2];
+
+    return ESP_OK;
+}
+```
+
+Now that we have the MPU driver implementation, let's dive into the main component implementation!
+
+
+
+### Main component implementation
+
+The main component is in charge of initializing the sensor and periodically print sensor values on the screen.  
+The content of `mpu65000-tester.c` could be:
+
+```C
+#include <accelerometer-driver.h>
+#include <esp_log.h>
+#include <esp_err.h>
+#include <freertos/FreeRTOS.h>
+#include <freertos/task.h>
+
+static const char* TAG = "mpu6500-tester";
+
+
+void app_main() {
+    mpu6500_sensor_t *sensor = NULL;
+
+    ESP_ERROR_CHECK(mpu6500_sensor_init(&sensor));
+
+    while (1) {
+        float a_x, a_y, a_z, g_x, g_y, g_z;
+        ESP_ERROR_CHECK(accelerometer_read(sensor, &a_x, &a_y, &a_z));
+        ESP_LOGI(TAG, "A (x, y, z) : (%+6.2f, %+6.2f, %+6.2f)", a_x, a_y, a_z);
+        ESP_ERROR_CHECK(gyroscope_read(sensor, &g_x, &g_y, &g_z));
+        ESP_LOGI(TAG, "G (x, y, z) : (%+6.2f, %+6.2f, %+6.2f)", g_x, g_y, g_z);
+
+        vTaskDelay(pdMS_TO_TICKS(4000));
+    }
+}
+```
+
+
+
+### Build and flash
+
+Before build and flash the project, you have to select the correct MPU model. To configure the project, open the `menuconfig` tool typing in the rott folder project:
```

Comment:
Primary sentence first:
`You have to select the correct MPU model before building and flashing the project.`